src/main/java/com/digirati/taxonomy/manager/lookup/ConceptLookupHandler.java

```diff
@@ -0,0 +1,54 @@
+package com.digirati.taxonomy.manager.lookup;
+
+import com.digirati.taxonomy.manager.lookup.model.Concept;
+import com.digirati.taxonomy.manager.lookup.model.ConceptMatch;
+import com.digirati.taxonomy.manager.lookup.model.TermMatch;
+import com.google.common.collect.ArrayListMultimap;
+import com.google.common.collect.Multimap;
+
+import java.util.Collection;
+import java.util.stream.Collectors;
+
+/**
+ * Encapsulates all operations related to determining which concepts exist within a given piece of
+ * text.
+ */
+public class ConceptLookupHandler {
```

Comment:
Hm, it does do concept lookups for a given text, but the "Handler" suffix gives the impression that it's a sort of event listener, or reacts to a request.

What do you think about `TextClassifier`, or `TextCategorizer` (and associated `ConceptLookupHandler#search` -> `TextClassifier|Categorier#classify|categorize`)?

cc @p-kaczynski @danielgrant for bikeshedding some names.

Reply:
I couldn't think of a good name in the moment (hence this not great one) but now that I've pushed it up I've just had a thought - how about `ConceptExtractor#extract`?

Reply:
Yep, that sounds good to me.